packages/remix-dev/cli/commands.ts

```diff
@@ -60,7 +60,11 @@ export async function build(
 
   let start = Date.now();
   let config = await readConfig(remixRoot);
-  await compiler.build(config, { mode: mode, sourcemap });
+  try {
+    await compiler.build(config, { mode: mode, sourcemap });
+  } catch (error: unknown) {
```

Comment:
I was under the impression that a runtime error would result in a non-zero exit code.

Reply:
They get swallowed here https://github.com/remix-run/remix/blob/197f570841e4813b2005d1ae007523c09df3f5c5/packages/remix-dev/compiler.ts#L307-L314